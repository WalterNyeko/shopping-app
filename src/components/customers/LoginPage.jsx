import React from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import useStyles from '../../styles/Signup';

const LoginPage = ({state: {email, password}, handleInputChange, handleSubmit }) => {
  const classes = useStyles();
  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
      <Typography component="h1" variant="h5">
          ShopMate
        </Typography>
        <form className={classes.form} noValidate>
        
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                id="email"
                label="Email Address"
                name="email"
                autoComplete="email"
                defaultValue={email}
                onChange={handleInputChange}
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                name="password"
                label="Password"
                type="password"
                id="password"
                autoComplete="current-password"
                defaultValue={password}
                onChange={handleInputChange}
              />
            </Grid>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={handleSubmit}
          >
            Sign In
          </Button>
        </form>
      </div>
    </Container>
  );
}
export default LoginPage;