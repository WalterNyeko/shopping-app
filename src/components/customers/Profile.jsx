import React from "react";

export const Profile = ({
  theCustomer,
  handleInputChange,
  state,
  hanleEditCreditCard,
  hanleEditAddress,
  hanleEditProfile,
  shippingRegions,
  currentRegion
}) => {
  const {
    name,
    email,
    address_1,
    address_2,
    city,
    region,
    postal_code,
    country,
    shipping_region_id,
    credit_card,
    day_phone,
    eve_phone,
    mob_phone
  } = theCustomer.customerDetails;

  return (
    <section className="container content mb-5">
      <div className="row">
        <div className="col-md-5">
          <div className="card">
            <div className="card-body">
              <div className="user-image text-center mb-4">
                <i className="fa fa-user-circle fa-8x" />
              </div>
              <div className="profile">
                <ul className="list-group">
                  <li className="list-group-item no-line">
                    <strong>Full Name:</strong>
                    <span className="float-right">{name ? name : ""}</span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Email:</strong>
                    <span className="float-right">{email ? email : ""}</span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Day Phone:</strong>
                    <span className="float-right">
                      {day_phone ? day_phone : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Evening Phone:</strong>
                    <span className="float-right">
                      {eve_phone ? eve_phone : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Mobile Phone:</strong>
                    <span className="float-right">
                      {mob_phone ? mob_phone : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <p>
                      <a
                        className="btn btn-danger btn-sm"
                        data-toggle="collapse"
                        href="#multiCollapseExample1"
                        role="button"
                        aria-expanded="false"
                        aria-controls="multiCollapseExample1"
                      >
                        Edit <i className="fas fa-pen" />
                      </a>
                    </p>
                    <div className="row">
                      <div className="col">
                        <div
                          className="collapse multi-collapse"
                          id="multiCollapseExample1"
                        >
                          <form>
                            <div className="form-group">
                              <label>Full Name</label>
                              <input
                                type="text"
                                className="form-control"
                                defaultValue={state.name}
                                onChange={handleInputChange}
                                name="name"
                              />
                              {state.errors.name && (
                                <span className="text-danger">
                                  {state.errors.name}
                                </span>
                              )}
                            </div>
                            <div className="form-group">
                              <label>Email</label>
                              <input
                                type="text"
                                className="form-control"
                                defaultValue={state.email}
                                onChange={handleInputChange}
                                name="email"
                              />
                              {state.errors.email && (
                                <span className="text-danger">
                                  {state.errors.email}
                                </span>
                              )}
                            </div>
                            <div className="form-group">
                              <label>Password</label>
                              <input
                                type="password"
                                className="form-control"
                                defaultValue={state.password}
                                onChange={handleInputChange}
                                name="password"
                              />
                            </div>
                            <div className="form-group">
                              <label>Day Phone</label>
                              <input
                                type="text"
                                className="form-control"
                                defaultValue={state.day_phone}
                                onChange={handleInputChange}
                                name="day_phone"
                              />
                            </div>
                            <div className="form-group">
                              <label>Evening Phone</label>
                              <input
                                type="text"
                                className="form-control"
                                defaultValue={state.eve_phone}
                                onChange={handleInputChange}
                                name="eve_phone"
                              />
                            </div>
                            <div className="form-group">
                              <label>Mobile Phone</label>
                              <input
                                type="text"
                                className="form-control"
                                defaultValue={state.mob_phone}
                                onChange={handleInputChange}
                                name="mob_phone"
                              />
                            </div>
                            <div className="form-group">
                              <button
                                type="submit"
                                className="btn btn-primary update-profile"
                                onClick={hanleEditProfile}
                              >
                                Update
                              </button>
                            </div>
                          </form>
                        </div>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div className="col-md-7">
          <div className="address">
            <div className="card">
              <div className="card-header">Address</div>
              <div className="card-body">
                <ul className="list-group">
                  <li className="list-group-item no-line">
                    <strong>Address 1:</strong>
                    <span className="float-right">
                      {address_1 ? address_1 : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Address 2:</strong>
                    <span className="float-right">
                      {address_2 ? address_2 : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>City:</strong>
                    <span className="float-right">{city ? city : ""}</span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Region:</strong>
                    <span className="float-right">{region ? region : ""}</span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Postal Code:</strong>
                    <span className="float-right">
                      {postal_code ? postal_code : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Counrty:</strong>
                    <span className="float-right">
                      {country ? country : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <strong>Shipping Region:</strong>
                    <span className="float-right">
                      {shipping_region_id ? currentRegion : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <p>
                      <a
                        className="btn btn-danger btn-sm"
                        data-toggle="collapse"
                        href="#multiCollapseExample2"
                        role="button"
                        aria-expanded="false"
                        aria-controls="multiCollapseExample2"
                      >
                        Edit <i className="fas fa-pen" />
                      </a>
                    </p>
                    <div className="row">
                      <div className="col">
                        <div
                          className="collapse multi-collapse"
                          id="multiCollapseExample2"
                        >
                          <form>
                            <div className="row">
                              <div className="form-group col-md-6">
                                <label>Address 1</label>
                                <input
                                  type="text"
                                  className="form-control"
                                  defaultValue={state.address_1}
                                  onChange={handleInputChange}
                                  name="address_1"
                                />
                                {state.errors.address_1 && (
                                  <span className="text-danger">
                                    {state.errors.address_1}
                                  </span>
                                )}
                              </div>
                              <div className="form-group col-md-6">
                                <label>Address 2</label>
                                <input
                                  type="text"
                                  className="form-control"
                                  defaultValue={state.address_2}
                                  onChange={handleInputChange}
                                  name="address_2"
                                />
                              </div>
                            </div>

                            <div className="row">
                              <div className="form-group col-md-6">
                                <label>City</label>
                                <input
                                  type="text"
                                  className="form-control"
                                  defaultValue={state.city}
                                  onChange={handleInputChange}
                                  name="city"
                                />
                                {state.errors.city && (
                                  <span className="text-danger">
                                    {state.errors.city}
                                  </span>
                                )}
                              </div>
                              <div className="form-group col-md-6">
                                <label>Region</label>
                                <input
                                  type="text"
                                  className="form-control"
                                  defaultValue={state.region}
                                  onChange={handleInputChange}
                                  name="region"
                                />
                                {state.errors.region && (
                                  <span className="text-danger">
                                    {state.errors.region}
                                  </span>
                                )}
                              </div>
                            </div>
                            <div className="row">
                              <div className="form-group col-md-6">
                                <label>Postal Code</label>
                                <input
                                  type="text"
                                  className="form-control"
                                  defaultValue={state.postal_code}
                                  onChange={handleInputChange}
                                  name="postal_code"
                                />
                                {state.errors.postal_code && (
                                  <span className="text-danger">
                                    {state.errors.postal_code}
                                  </span>
                                )}
                              </div>
                              <div className="form-group col-md-6">
                                <label>Country</label>
                                <input
                                  type="text"
                                  className="form-control"
                                  defaultValue={state.country}
                                  onChange={handleInputChange}
                                  name="country"
                                />
                                {state.errors.country && (
                                  <span className="text-danger">
                                    {state.errors.country}
                                  </span>
                                )}
                              </div>
                            </div>
                            <div className="row">
                              <div className="form-group col-md-6">
                                <label>Shipping Region ID</label>
                                <select
                                  className="form-control"
                                  onChange={handleInputChange}
                                  name="shipping_region_id"
                                >
                                  {shippingRegions &&
                                    shippingRegions.length &&
                                    shippingRegions.map(
                                      ({
                                        index,
                                        shipping_region_id,
                                        shipping_region
                                      }) => (
                                        <option
                                          value={shipping_region_id}
                                          key={index}
                                        >
                                          {shipping_region}
                                        </option>
                                      )
                                    )}
                                </select>
                                {state.errors.shipping_region_id && (
                                  <span className="text-danger">
                                    {state.errors.shipping_region_id}
                                  </span>
                                )}
                              </div>
                              <div className="form-group col-md-6 mt-2">
                                <label />
                                <button
                                  type="submit"
                                  className="btn btn-primary mt-4 update-address"
                                  onClick={hanleEditAddress}
                                >
                                  Update
                                </button>
                              </div>
                            </div>
                          </form>
                        </div>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>

          <div className="credit-card mt-4">
            <div className="card">
              <div className="card-header">Credit Card</div>
              <div className="card-body">
                <ul className="list-group">
                  <li className="list-group-item no-line">
                    <strong>Card Number:</strong>
                    <span className="float-right">
                      {credit_card ? credit_card : ""}
                    </span>
                  </li>
                  <li className="list-group-item no-line">
                    <p>
                      <a
                        className="btn btn-danger btn-sm"
                        data-toggle="collapse"
                        href="#multiCollapseExample3"
                        role="button"
                        aria-expanded="false"
                        aria-controls="multiCollapseExample3"
                      >
                        Edit <i className="fas fa-pen" />
                      </a>
                    </p>
                    <div className="row">
                      <div className="col">
                        <div
                          className="collapse multi-collapse"
                          id="multiCollapseExample3"
                        >
                          <form>
                            <div className="form-group">
                              <label>Card Number</label>
                              <input
                                type="text"
                                className="form-control credit_card"
                                defaultValue={credit_card ? credit_card : ""}
                                onChange={handleInputChange}
                                name="credit_card"
                              />
                              {state.errors.credit_card && (
                                <span className="text-danger">
                                  {state.errors.credit_card}
                                </span>
                              )}
                            </div>
                            <div className="form-group">
                              <button
                                type="submit"
                                className="btn btn-primary  update-card"
                                onClick={hanleEditCreditCard}
                              >
                                Update
                              </button>
                            </div>
                          </form>
                        </div>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Profile;
