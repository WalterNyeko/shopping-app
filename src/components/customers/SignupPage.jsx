import React from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import useStyles from '../../styles/Signup';

const SignUp = ({ handleSubmit, handleInputChange, 
  state: {username, email, password}}) => {
  const classes = useStyles();
  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
      <Typography component="h1" variant="h5">
          ShopMate
        </Typography>
        <form className={classes.form} noValidate>
        
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <TextField
                autoComplete="username"
                name="username"
                variant="outlined"
                required
                fullWidth
                id="username"
                label="Username"
                defaultValue={username}
                onChange={handleInputChange}
                autoFocus
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                id="email"
                label="Email Address"
                defaultValue={email}
                onChange={handleInputChange}
                name="email"
                autoComplete="email"
              />
            </Grid>
            <Grid item xs={12}>
              <TextField
                variant="outlined"
                required
                fullWidth
                name="password"
                label="Password"
                defaultValue={password}
                onChange={handleInputChange}
                type="password"
                id="password"
                autoComplete="current-password"
              />
            </Grid>
          </Grid>
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
            onClick={handleSubmit}
          >
            Sign Up
          </Button>
        </form>
      </div>
    </Container>
  );
}
export default SignUp;
