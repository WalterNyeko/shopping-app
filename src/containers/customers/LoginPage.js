import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import LoginPageComponent from "../../components/customers/LoginPage";
import { signIn } from "../../store/actions/Customers";

export class LoginPage extends Component {
  state = {
    email: "",
    password: ""
  };

  /**
   * ensures the most recent value for each jsx input element is stored in the state
   * sets the state value of the input basing on its name attribute
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  /**
   * signs in a user into the app upon successful request
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleSubmit = event => {
    event.preventDefault();
    const { signIn } = this.props;
    const { email, password } = this.state;
    const userData = { email, password };
    signIn(userData);
  };
  render() {
    return (
      <Fragment>
        <LoginPageComponent
          state={this.state}
          handleInputChange={this.handleInputChange}
          handleSubmit={this.handleSubmit}
        />
      </Fragment>
    );
  }
}
export default connect(
  null,
  { signIn }
)(LoginPage);
