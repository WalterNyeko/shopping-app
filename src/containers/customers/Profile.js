import React, { Fragment } from "react";
import { connect } from "react-redux";
import ProfileComponent from "../../components/customers/Profile";
import UpperNav from "../../components/commons/UpperNav";
import Footer from "../../components/commons/Footer";
import { getDepartments } from "../../store/actions/Departments";
import {
  getCustomersDetails,
  updateProfile,
  updateAddress,
  updateCreditCard,
  getShippingRegions
} from "../../store/actions/Customers";
import history from "../../helpers/history";

export class Profile extends React.Component {
  state = {
    isLoggedIn: false,
    name: "",
    email: "",
    address_1: "",
    address_2: "",
    city: "",
    region: "",
    postal_code: "",
    country: "",
    shipping_region_id: "",
    credit_card: "",
    day_phone: "",
    eve_phone: "",
    mob_phone: "",
    password: "",
    errors: {}
  };
  componentWillMount() {
    const {
      getDepartments,
      getCustomersDetails,
      getShippingRegions
    } = this.props;
    getDepartments();
    const jwtToken = localStorage.getItem("jwt-token");
    jwtToken && this.setState({ isLoggedIn: true });
    jwtToken && getCustomersDetails();
    getShippingRegions();
  }

  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  hanleEditProfile = event => {
    event.preventDefault();
    const { updateProfile } = this.props;
    const {
      name,
      email,
      password,
      day_phone,
      eve_phone,
      mob_phone
    } = this.state;
    const data = {
      name,
      email,
      password,
      day_phone,
      eve_phone,
      mob_phone
    };
    if (!name) {
      return this.inputErrors("name", "Please provide your name");
    } else if (!email) {
      return this.inputErrors("email", "Please provide your email address");
    } else {
      updateProfile(data);
    }
  };

  hanleEditAddress = event => {
    event.preventDefault();
    const { updateAddress } = this.props;
    const data = {
      address_1: this.state.address_1,
      address_2: this.state.address_2,
      city: this.state.city,
      region: this.state.region,
      postal_code: this.state.postal_code,
      country: this.state.country,
      shipping_region_id: this.state.shipping_region_id
    };
    if (!this.state.address_1) {
      return this.inputErrors("address_1", "Field address 1 is mandatory");
    } else if (!this.state.city) {
      return this.inputErrors("city", "Please provide your city");
    } else if (!this.state.region) {
      return this.inputErrors("region", "Please provide your region");
    } else if (!this.state.postal_code) {
      return this.inputErrors("postal_code", "Please provide your postal code");
    } else if (!this.state.country) {
      return this.inputErrors("country", "Please provide your country");
    } else if (!this.state.shipping_region_id) {
      return this.inputErrors(
        "shipping_region_id",
        "Please select your shipping region"
      );
    } else {
      updateAddress(data);
    }
  };

  hanleEditCreditCard = event => {
    event.preventDefault();
    const { updateCreditCard } = this.props;
    const { credit_card } = this.state;
    const data = {
      credit_card
    };
    if (!credit_card) {
      return this.inputErrors(
        "credit_card",
        "Please provide your credit card number"
      );
    } else {
      updateCreditCard(data);
    }
  };

  handleClickNav = id => {
    history.push(`/department/${id}`);
  };

  /**
   * renders the validation errors for input fields
   *
   * @param {string} errorKey
   * @param {string} message
   *
   * @returns {void}
   */
  inputErrors = (errorKey, message) => {
    this.setState({
      ...(this.state.errors[errorKey] = message)
    });
  };

  /**
   * logs out a user from the app by doing the following
   * 1. clears the users token from localStorage
   * 2. clears the users name from loaclStorage
   * 3. redirects user to the landing page
   *
   * @returns {void}
   */
  handleLogout = () => {
    localStorage.removeItem("jwt-token");
    localStorage.removeItem("user");
    const { pathname } = history.location;
    if (pathname === "/") {
      history.push("/home");
    } else {
      history.push("/");
    }
  };

  render() {
    const {
      allDepartments,
      allDepartments: { departments },
      theCustomer,
      theCustomer: { shippingRegions }
    } = this.props;
    const { isLoggedIn } = this.state;
    const { shipping_region_id } = theCustomer.customerDetails;
    const theRegion = shippingRegions.filter(
      element => element.shipping_region_id === shipping_region_id
    );
    let currentRegion;
    if (theRegion.length) {
      currentRegion = theRegion[0].shipping_region;
    }

    return (
      <Fragment>
        <UpperNav
          allDepartments={allDepartments}
          isLoggedIn={isLoggedIn}
          handleClickNav={this.handleClickNav}
          handleLogout={this.handleLogout}
        />
        <ProfileComponent
          theCustomer={theCustomer}
          shippingRegions={shippingRegions}
          handleInputChange={this.handleInputChange}
          state={this.state}
          hanleEditProfile={this.hanleEditProfile}
          hanleEditAddress={this.hanleEditAddress}
          hanleEditCreditCard={this.hanleEditCreditCard}
          currentRegion={currentRegion}
        />
        <Footer departments={departments} />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  allDepartments: state.departments,
  theCustomer: state.customers
});
export default connect(
  mapStateToProps,
  {
    getDepartments,
    getCustomersDetails,
    updateProfile,
    updateAddress,
    updateCreditCard,
    getShippingRegions
  }
)(Profile);
