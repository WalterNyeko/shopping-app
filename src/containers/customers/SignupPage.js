import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import SignupPageComponent from "../../components/customers/SignupPage";
import { signupUser } from "../../store/actions/Customers";

export class SignupPage extends Component {
  state = {
    username: "",
    email: "",
    password: ""
  };

  /**
   * ensures the most recent value for each jsx input element is stored in the state
   * sets the state value of the input basing on its name attribute
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  /**
   * signs up a user into the app upon successful request
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleSubmit = event => {
    event.preventDefault();
    const { username, email, password } = this.state;
    const userData = { name: username, email, password };
    const { signupUser } = this.props;
    signupUser(userData);
  };
  render() {
    return (
      <Fragment>
        <SignupPageComponent
          handleSubmit={this.handleSubmit}
          handleInputChange={this.handleInputChange}
          state={this.state}
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  username: state.username,
  email: state.email,
  password: state.password
});
export default connect(
  mapStateToProps,
  { signupUser }
)(SignupPage);
