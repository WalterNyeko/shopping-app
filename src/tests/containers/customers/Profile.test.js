import React from "react";
import { shallow } from "enzyme";
import { Profile as ProfileComponent } from "../../../components/customers/Profile";
import { Profile } from "../../../containers/customers/Profile";

describe("Profile", () => {
  const props = {
    getDepartments: jest.fn(),
    getShippingRegions: jest.fn(),
    getCustomersDetails: jest.fn(),
    updateCreditCard: jest.fn(),
    allDepartments: {
      departments: [{}]
    },
    theCustomer: {
      shippingRegions: [{ shipping_region_id: 1 }],
      customerDetails: { shipping_region_id: 1 }
    },
    state: {
      isLoggedIn: false,
      name: "",
      email: "",
      address_1: "",
      address_2: "",
      city: "",
      region: "",
      postal_code: "",
      country: "",
      shipping_region_id: "",
      credit_card: "",
      day_phone: "",
      eve_phone: "",
      mob_phone: "",
      password: "",
      errors: {}
    },
    updateProfile: jest.fn(),
    updateAddress: jest.fn()
  };
  const wrapper = shallow(<Profile {...props} />);
  it("should render profile correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call inputErrors() when validation for credit card does not pass", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateCardButton = componentWrapper.find(".update-card");
    const event = {
      preventDefault: jest.fn()
    };
    const spy = jest.spyOn(wrapper.instance(), "inputErrors");
    updateCardButton.simulate("click", event);
    expect(spy).toHaveBeenCalled();
  });

  it("should set state value correctly when inputErrors() is called", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateCardButton = componentWrapper.find(".update-card");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({ credit_card: "" });
    const spy = jest.spyOn(wrapper.instance(), "inputErrors");
    updateCardButton.simulate("click", event);
    expect(spy).toHaveBeenCalled();
    expect(wrapper.instance().state.errors.credit_card).toEqual(
      "Please provide your credit card number"
    );
  });

  it("should call updateCreditCard() when validations pass while editing Card", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateCardButton = componentWrapper.find(".update-card");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.instance().handleClickNav(1);
    wrapper.instance().handleLogout();
    wrapper.setState({ credit_card: "42424242424242" });
    const spy = jest.spyOn(wrapper.instance().props, "updateCreditCard");
    updateCardButton.simulate("click", event);
    expect(spy).toHaveBeenCalled();
  });

  it("should call updateProfile() when validations pass while editing Profile", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateProfileButton = componentWrapper.find(".update-profile");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({ name: "nyeko", email: "nyeko@gmail.com" });
    const spy = jest.spyOn(wrapper.instance().props, "updateProfile");
    updateProfileButton.simulate("click", event);
    expect(spy).toHaveBeenCalled();
  });

  it("should call updateAddress() when validations pass while editing Address", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: "nyeko",
      city: "nyeko@gmail.com",
      region: "West Coast",
      postal_code: "256",
      country: "uganda",
      shipping_region_id: 1
    });
    const spy = jest.spyOn(wrapper.instance().props, "updateAddress");
    updateAddressButton.simulate("click", event);
    expect(spy).toHaveBeenCalled();
  });

  it("should call handleInputChange() and update state when input values change", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const creditCardField = componentWrapper.find(".credit_card");
    const event = {
      preventDefault: jest.fn(),
      target: { name: "credit_card", value: "4242424242424" }
    };
    creditCardField.simulate("change", event);
    expect(wrapper.instance().state.credit_card).toEqual("4242424242424");
  });

  it("should call inputErrors() and update errors state when value for address_1 is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: ""
    });
    updateAddressButton.simulate("click", event);
    expect(wrapper.instance().state.errors.address_1).toEqual(
      "Field address 1 is mandatory"
    );
  });

  it("should call inputErrors() and update errors state when value for city is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: "test",
      city: ""
    });
    updateAddressButton.simulate("click", event);
    expect(wrapper.instance().state.errors.city).toEqual(
      "Please provide your city"
    );
  });

  it("should call inputErrors() and update errors state when value for region is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: "test",
      city: "test",
      region: ""
    });
    updateAddressButton.simulate("click", event);
    expect(wrapper.instance().state.errors.region).toEqual(
      "Please provide your region"
    );
  });

  it("should call inputErrors() and update errors state when value for postal_code is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: "test",
      city: "test",
      region: "test",
      postal_code: ""
    });
    updateAddressButton.simulate("click", event);
    expect(wrapper.instance().state.errors.postal_code).toEqual(
      "Please provide your postal code"
    );
  });

  it("should call inputErrors() and update errors state when value for country is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: "test",
      city: "test",
      region: "test",
      postal_code: "256",
      country: ""
    });
    updateAddressButton.simulate("click", event);
    expect(wrapper.instance().state.errors.country).toEqual(
      "Please provide your country"
    );
  });

  it("should call inputErrors() and update errors state when value for shipping_region_id is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateAddressButton = componentWrapper.find(".update-address");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      address_1: "test",
      city: "test",
      region: "test",
      postal_code: "256",
      country: "test",
      shipping_region_id: ""
    });
    updateAddressButton.simulate("click", event);
    expect(wrapper.instance().state.errors.shipping_region_id).toEqual(
      "Please select your shipping region"
    );
  });

  it("should call inputErrors() and update errors state when value for name is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateProfileButton = componentWrapper.find(".update-profile");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      name: ""
    });
    updateProfileButton.simulate("click", event);
    expect(wrapper.instance().state.errors.name).toEqual(
      "Please provide your name"
    );
  });

  it("should call inputErrors() and update errors state when value for email is missing", () => {
    const componentWrapper = wrapper.find(ProfileComponent).dive();
    const updateProfileButton = componentWrapper.find(".update-profile");
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      name: "test",
      email: ""
    });
    updateProfileButton.simulate("click", event);
    expect(wrapper.instance().state.errors.email).toEqual(
      "Please provide your email address"
    );
  });
});
// wrapper.setState({
//     address_1: "",
//     city: "",
//     region: "",
//     postal_code: "",
//     country: "",
//     shipping_region_id: ""
//   });
