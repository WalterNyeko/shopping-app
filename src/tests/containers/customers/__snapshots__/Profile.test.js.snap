// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render profile correctly 1`] = `ShallowWrapper {}`;
